### PR TITLE
Numerous fixes and such

### PR DESCRIPTION
**Pull request by kitskub**
Fix tp command
Fix the variable naming in LoginMessage to reflect what it actually is
Changed NOTSPAWNEDID to be -999 because -1 was/is being used as a player id (this needs polishing)
Refactored start(boolean) in SpoutEngine to just start()
Fix my session pulsing error from last PR

Signed-off-by: Jack Huey kitskub@gmail.com

### CODE REVIEW

File: src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java [hunk 1]
```diff
@@ -33,26 +33,26 @@
 
 public class LoginMessage extends SpoutMessage {
 	private final String playerName;
-	private final int protoVersion;
```
Comment (zml2008) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: It's actually used as both, protocol version C->S and player id S->C

**Comment by windy1** [comment]
Why does this appear to be amended onto @Afforess's commit?

File: src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java (hunk 1 above)
Comment (kitskub) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: Hmm. You're right: the client is putting the protocol version in the message. But the server isn't using it.

Comment (zml2008) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: I don't think I ever got around to implementing protocol version checking, which is probably why.

Comment (kitskub) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: Well, I went ahead and named it appropriately (extraInt)

**Comment by kitskub** [comment]
@W1ndwaker I think I accidently amended the commit. Fixed.

File: src/main/java/org/spout/engine/entity/SpoutEntity.java [hunk 2]
```diff
@@ -73,14 +73,14 @@
 import org.spout.engine.world.SpoutRegion;
 
 public class SpoutEntity extends BaseComponentOwner implements Entity, Snapshotable {
-	public static final int NOTSPAWNEDID = -1;
+	public static final int NOTSPAWNEDID = -999;
```
Comment (windy1) [inline src/main/java/org/spout/engine/entity/SpoutEntity.java]: If we're allowing negative Entity IDs then why wouldn't we use `Integer.MIN_VALUE`

File: src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java (hunk 1 above)
Comment (windy1) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: "extraInt" is incredibly ambiguous.

Comment (kitskub) [inline src/main/java/org/spout/engine/protocol/builtin/message/LoginMessage.java:36]: Well, I COULD make it playerIdOrProtocolVersion?

File: src/main/java/org/spout/engine/SpoutClient.java [hunk 3]
```diff
@@ -226,6 +221,16 @@ private boolean connnect() {
 	}
 
 	@Override
+	protected Runnable getSessionTask() {
+		return new Runnable() {
```
Comment (Afforess) [inline src/main/java/org/spout/engine/SpoutClient.java]: One minor suggestion. Make this a private final variable, and create the runnable once. Then return the variable instead. The outcome is the same, with the advantage of not creating an object every tick.

File: src/main/java/org/spout/engine/SpoutServer.java [hunk 4]
```diff
@@ -320,17 +315,14 @@ public SessionRegistry getSessionRegistry() {
...
-			this.registry = registry;
-		}
-
-		@Override
-		public void run() {
-			registry.pulse();
-		}
+	@Override
+	protected Runnable getSessionTask() {
+		return new Runnable() {
```
Comment (Afforess) [inline src/main/java/org/spout/engine/SpoutServer.java]: Same as earlier. Also, I had the code in a private class on purpose. When exceptions are thrown, it's much easier to pick out the cause with named inner classes versus the anonymous inner class names that are autogenerated.

**Comment by kitskub** [comment]
Fixed. In my defense, I thought when the task was scheduled, only one instance was created.

**Pull request merged**